pysmiles/smiles_helper.py

```diff
@@ -876,6 +877,42 @@ def _mark_chiral_atoms(molecule):
             neighbours = [neighbours[0],  neighbours[1], neighbours[3], neighbours[2]]
         molecule.nodes[node]['rs_isomer'] = tuple(neighbours)
 
+def check_for_ez_conflicts(anchor, tagged_nodes, ez_isomer_class):
+    n1, n2 = tagged_nodes
+    if (n1 < anchor and n2 < anchor) or (n1 > anchor and n2 > anchor):
+        assert ez_isomer_class[n1] != ez_isomer_class[n2]
+    else:
+        assert ez_isomer_class[n1] == ez_isomer_class[n2]
+
+def annotate_ez_isomers(molecule, ez_atoms):
+    """
+    Classify E/Z isomers atoms as cis or trans.
+    """
+    ez_isomer_pairs = []
+    for anchor1, anchor2, order in molecule.edges(data='order'):
+        if order != 2:
+            continue
+        if anchor1 not in ez_atoms or anchor2 not in ez_atoms:
+            if anchor1 in ez_atoms or anchor2 in ez_atoms:
+                msg = (f"Dangling E/Z isomer token for double bond between {anchor1} {anchor2}.")
+                raise ValueError(msg)
+            continue
+        anchors = [anchor1, anchor2]
+        # we have a double bond and need to check cis/trans
+        ez_on_anchor = defaultdict(list)
+        for anchor in anchors:
+            tagged_nodes = []
+            for neighbor in molecule.neighbors(anchor):
+                # we have a relation
+                if neighbor in ez_atoms and neighbor not in anchors:
```

Comment:
Nit. I think this makes the intent a little more clear
```suggestion
            for neighbor in set(molecule.neighbors(anchor)) - set(anchors):
                # we have a relation
                if neighbor in ez_atoms:
```